### PR TITLE
User / passwd handling and temp User fix

### PR DESCRIPTION
**Pull request by chhex**
Following changes: 
1. Password Prompt via Apgsecrets gem https://github.com/apgsga-it/apg-ruby-gems/tree/main/apgsecrets 
2. Temp User Parameter
3.  User with which git.apgsga.ch is accessed 

First change:
To allow unattended execution of the bolt plans, instead of each plan prompted for the same password
With the apgsecrets the password is effectively very short time in memory and stored encrypted in a temporary file storage, which expires after 7200 sec. 

Second change: 
Plans, which until now how taken the -u user from target[ssh][user] are provided with a user parameter. 
Is will be replaced in further iterations with prefined users 

Third change: 
It is assumed that the -u user is the same user with which git.apgsga.ch is accessed

### CODE REVIEW

File: patchserver-setup.rb [hunk 1]
```diff
@@ -22,6 +23,10 @@
     exit
   end
 end
+plans_with_user_param = []
```
Comment (chhex) [inline patchserver-setup.rb:26]: Tempory : Plans which need a user parameter , which is actually the same as the -u user parameter

File: patchserver-setup.rb [hunk 2]
```diff
@@ -48,15 +53,16 @@
...
 if !opts[:skipClone]
   bolt_inventory_file = File.join(File.dirname(__FILE__), 'inventory.yaml')
   inventory = YAML.load_file(bolt_inventory_file)
   temp_dir = inventory['vars']['temp_gradle']
   if  File.exist?(temp_dir)
     FileUtils.remove_dir(temp_dir, force = true)
   end
-  # TODO JHE : git user, currently hardcoded with jhe -> IT-36770 and IT-36776
-  system("git clone jhe@git.apgsga.ch:/var/git/repos/apg-gradle-properties.git #{temp_dir}")
+  system("git clone #{opts[:user]}@git.apgsga.ch:/var/git/repos/apg-gradle-properties.git #{temp_dir}")
```
Comment (chhex) [inline patchserver-setup.rb:65]: It is assumed the user with which git.apgsga.ch is accessed is the same as the -u user

File: patchserver-setup.rb [hunk 3]
```diff
@@ -88,11 +94,12 @@
   plans_to_execute = plans
 end
 
-def run(plan,opts)
+def run(plan,opts,secrets,parm)
```
Comment (chhex) [inline patchserver-setup.rb:97]: Temporary User Parameter

File: site-modules/piper/plans/jenkins_create_jobs.pp [hunk 4]
```diff
@@ -1,5 +1,6 @@
 plan piper::jenkins_create_jobs (
-     TargetSpec $targets
+     TargetSpec $targets,
+     TargetSpec $user
```
Comment (chhex) [inline site-modules/piper/plans/jenkins_create_jobs.pp:3]: Temporary user parameter

File: site-modules/piper/plans/jenkins_dirs_create.pp [hunk 5]
```diff
@@ -1,5 +1,6 @@
 plan piper::jenkins_dirs_create (
-  TargetSpec $targets
+  TargetSpec $targets,
+  String $user
```
Comment (chhex) [inline site-modules/piper/plans/jenkins_dirs_create.pp:3]: Temporary user parameter

File: site-modules/piper/plans/piper_service_properties.pp [hunk 6]
```diff
@@ -1,5 +1,6 @@
 plan piper::piper_service_properties (
-     TargetSpec $targets
+     TargetSpec $targets,
+     String $user
```
Comment (chhex) [inline site-modules/piper/plans/piper_service_properties.pp:3]: Ditto, temporary user parameter

**Comment by chhex** [comment]
@apgsga-uge @apgsga-jhe This pull request is more informative, i will merge after our next meeting. 
But feel free to review in any case.  
I can take the requests into the next iterations.

**Review by apgsga-jhe — APPROVED** [review]
@chhex  : Looking good to me.

File: inventory-local.yaml [hunk 7]
```diff
@@ -0,0 +1,19 @@
+name: local
```
Comment (chhex) [inline inventory-local.yaml:1]: As discussed the target group local, test & prod have been split into different files

File: inventory-local.yaml [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+targets:
+  - centpatch
```
Comment (chhex) [inline inventory-local.yaml:3]: Currently therefore the inventory-local.yaml still needs to be adopted to reflect the local configuration requirements.

File: inventory.yaml [hunk 9]
```diff
@@ -1,29 +1,18 @@
...
-        piper_spring_profiles: live,remotecvs, patchOMock
-        piper_cvs_user: che # TODO (che, 7.1) -> svcCvsClient
-        jenkins_ssh_user: che # TODO (che, 7.1)  -> apg-patch-service-server
-        install_ssh_user: # TODO (che, 07.1)  -> apg_install
-        artifactory_repo: devche-repo
-        cm_db_host: chti212.apgsga.ch
-        cm_db_instance: chti212
-        cm_db_piper_user: cm_example_user
-        cm_db_piper_pw: cm_example_user_pw
+  - _plugin: yaml
```
Comment (chhex) [inline inventory.yaml:2]: The main inventory file now includes the files per target

**Comment by chhex** [comment]
Adopted patchserver-tests-setup.rb to the split inventory files: 

- The scripts reads now only from inventory.yaml and inventory-local.yaml
- New parameter -u resp --user.  Apgsecrets prompts and caches pw encrypted , like with the patchserver-setup.rb
- Moved testapp metadata from testappsconfig.yaml to inventory-local.yaml

**Comment by chhex** [comment]
@apgsga-jhe  I will merge soon, but please feel free to feedback ...

**Pull request merged**